wolox-books/src/app/components/navbar/navbar.component.ts

```diff
@@ -9,10 +14,20 @@ import { Router } from '@angular/router';
 })
 export class NavbarComponent {
 
-  constructor(private sessionService: SessionService, private router: Router) { }
+  quantity: number;
+
+  constructor(private sessionService: SessionService, private router: Router, private store: Store<AppState>) {
+    this.store.select("shoppingCart").subscribe(element => {
+      this.quantity = element['books'].length || 0;
+    });
```

Comment:
Si crees que la propiedad `books` existe y siempre tiene valor, entonces es suficiente haciendo `element['books'].length`, pero esto no valida que la propiedad exista o que tenga valor (si **no existe valor** tu code actual arrojará un *TypeError*); si piensas que el segundo escenario puede ocurrir, entonces debes pensar en algo como esto:

```suggestion
    this.store.select("shoppingCart").subscribe(({ books }) => {
      this.quantity = books && books.length ? books.length : 0;
    });
```